src/Uno.UI.Composition/Composition/ShapeVisual.skia.cs

```diff
@@ -74,10 +74,7 @@ private DrawingSession BeginShapesDrawing(in DrawingSession parentSession)
 			}
 			else
 			{
-				var shape = new SKPath();
-				shape.AddRect(new SKRect(viewBox.Offset.X, viewBox.Offset.Y, viewBox.Offset.X + viewBox.Size.X, viewBox.Offset.Y + viewBox.Size.Y));
-				shape.Transform(transform);
-				parentSession.Canvas.ClipPath(shape, antialias: true);
+				parentSession.Canvas.ClipRect(transform.MapRect(viewBox.GetRect()), antialias: true);
```

Comment:
@Youssef1313 I am curious, can we add any test to avoid/ or verify unnecessary SKPath allocations ?

Reply:
Not sure if there is a way to do so.